Source/Layout/BrickLayoutSection.swift

```diff
@@ -438,7 +437,95 @@ internal class BrickLayoutSection {
         return (startOrigin, maxY)
     }
 
+    func handleRow(for index: Int, maxHeight: CGFloat, updatedAttributes: OnAttributesUpdatedHandler?) {
+        if _dataSource.scrollDirection == .Vertical {
+            updateHeightAndAlignForRowWithStartingIndex(index, maxHeight: maxHeight, updatedAttributes: updatedAttributes)
+        }
+    }
+
+    /// Update the height for a row starting at a given index
+    func updateHeightAndAlignForRowWithStartingIndex(index: Int, maxHeight: CGFloat, updatedAttributes: OnAttributesUpdatedHandler?) {
+        guard index >= 0, let brickAttributes = self.attributes[index] else {
+            return
+        }
+
+        var currentIndex = index
+        let y = brickAttributes.originalFrame.origin.y
+        var rowWidthWithoutInsets: CGFloat = 0 // Keep track of all the widths, so we can calculate the correct inset for `Justified`-alignment
+        var rowAttributes: [BrickLayoutAttributes] = [] // Keep track of all attributes in this row
+
+        // Count down until attributes are found with a lower Y-origin
+        while currentIndex >= 0 {
+            guard let brickAttributes = attributes[currentIndex] where !brickAttributes.hidden else {
+                currentIndex -= 1
+                continue
+            }
+            if brickAttributes.originalFrame.origin.y != y {
+                break // Done, no more attributes on this row
+            }
+
+            rowAttributes.insert(brickAttributes, atIndex: 0) // insert at the front, so the attributes are sorted by lowest first
+            rowWidthWithoutInsets += brickAttributes.originalFrame.width
+            currentIndex -= 1
+        }
```

Comment:
This function is pretty large. Can we move the `rowAttributes` computation into a separate method? Would that be cleaner?